golem-ts-sdk/tests/testSetup.ts

```diff
@@ -1,18 +1,24 @@
-import { TypeMetadata } from '@golemcloud/golem-ts-types-core';
-import { Metadata } from '../.metadata/generated-types';
-import { TypescriptTypeRegistry } from '../src';
+import { vi } from 'vitest';
 
-// This setup is ran before every test suite (vitest worker)
-// and represents the entry point of any code-first user code
+// Global mocks which will be used within decorators,
+// These host functionalities shouldn't run when decorators run.
+// For example, getSelfMetadata is used in some decorators, however,
+// it executes only when `initiate` is called.
+// Also, these mocks are just place-holders. We can override the behavior
+// per tests using functionalities overrides module
```

Comment:
Initially i didn't want to do this much, but I need more unit tests for easier debugging when bumping into some errors during testing with templates